app/services/charts_layout_service.rb

```diff
@@ -31,9 +31,14 @@ def build_charts
 
   def find_chart_path
     if @fname
-      # First check if there is file for specific class of the @perf_record
+      # First, if this class specified a chart layout path, use it.
+      return chart_path(@layout, @perf_record.chart_layout_path) if
+        @perf_record.respond_to?(:chart_layout_path)
+
+      # Fallback to check if there is file for specific class of the @perf_record
       path = chart_path(@layout, @perf_record.class.name.gsub(/::/, '_'))
```

Comment:
If need be, a base class implementation could do this logic.  But, perhaps that makes more sense when we need to do more of this.